presto-main/src/test/java/com/facebook/presto/execution/TestThriftResourceGroupInfo.java

```diff
@@ -0,0 +1,351 @@
+
+    // Dummy values for fake QueryStateInfo
+    private static final QueryId FAKE_QUERY_ID_1 = new QueryId("20220202_215711_00938_kvz6g");
+    private static final QueryId FAKE_QUERY_ID_2 = new QueryId("20220202_215711_00938_ks4fq");
+    private static final ResourceGroupId FAKE_QUERY_RESOURCE_GROUP_ID = new ResourceGroupId(Arrays.asList("global", "pipeline", "ns", "sub_rg", "dummy_prefix", "dummy_query_task_id"));
+    private static final QueryState FAKE_QUERY_STATE = QueryState.DISPATCHING;
+    private static final String FAKE_QUERY = "SELECT something FROM somewhere";
+    private static final DateTime FAKE_CREATE_TIME = new DateTime("2000-01-01T01:00:00Z");
+    private static final String FAKE_QUERY_REQUESTER = "DummyUser";
+    private static final String FAKE_QUERY_SOURCE = "DummySource";
+    private static final String FAKE_QUERY_CLIENT_INFO = "DummyClientInfo";
+    private static final String FAKE_QUERY_CATALOG = "DummyCatalog";
+    private static final String FAKE_QUERY_SCHEMA = "DummySchema";
+
+    // Dummy values for fake QueryProgressStats
+    private static final long FAKE_ELAPSED_TIME_MILLIS = 10000L;
+    private static final long FAKE_QUEUED_TIME_MILLIS = 1000L;
+    private static final long FAKE_CPU_TIME_MILLS = 1L;
+    private static final long FAKE_SCHEDULED_TIME_MILLIS = 50L;
+    private static final long FAKE_CURRENT_MEMORY_BYTES = 10240L;
+    private static final long FAKE_PEAK_MEMORY_BYTES = 20480L;
+    private static final long FAKE_PEAK_TOTAL_MEMORY_BYTES = 40960L;
+    private static final long FAKE_PEAK_TASK_TOTAL_MEMORY_BYTES = 20480L;
+    private static final long FAKE_INPUT_ROWS = 5L;
+    private static final long FAKE_INPUT_BYTES = 1024L;
+    private static final boolean FAKE_BLOCKED = true;
+    private static final OptionalDouble FAKE_PROGRESS_PERCENTAGE_1 = OptionalDouble.of(65.23124);
+    private static final OptionalDouble FAKE_PROGRESS_PERCENTAGE_2 = OptionalDouble.of(98.235);
+
+    private ResourceGroupInfo resourceGroupInfo;
+    private List<QueryStateInfo> queryStateInfoList;
+    private List<QueryProgressStats> queryProgressStats;
+
+    @BeforeMethod
+    public void setUp()
+    {
+        setUpQueryProgressStats();
+        setUpQueryStateInfoList();
+        List<ResourceGroupInfo> subGroups = new ArrayList<>();
+        ResourceGroupInfo subResourceGroupInfo1 = getResourceGroupInfo(
+                FAKE_SUB_RESOURCE_GROUP_ID_1,
+                null,
+                null);
+        ResourceGroupInfo subResourceGroupInfo2 = getResourceGroupInfo(
+                FAKE_SUB_RESOURCE_GROUP_ID_2,
+                null,
+                null);
+        ResourceGroupInfo subResourceGroupInfo3 = getResourceGroupInfo(
+                FAKE_SUB_RESOURCE_GROUP_ID_3,
+                null,
+                null);
+        subGroups.add(subResourceGroupInfo1);
+        subGroups.add(subResourceGroupInfo2);
+        subGroups.add(subResourceGroupInfo3);
```

Comment:
move this to a method called createSubGroups

Reply:
Resolved locally. Will update in the next iter